pallets/connectors-gateway/src/lib.rs

```diff
@@ -274,91 +254,13 @@ pub mod pallet {
 			sender: Self::Sender,
 			msg: Self::Message,
 		) -> DispatchResult {
-			let Router::Xcm(xcm_domain) =
-				DomainRouters::<T>::get(destination).ok_or(Error::<T>::RouterNotFound)?;
-
-			let contract_call = encoded_contract_call(msg.serialize());
-			let ethereum_xcm_call =
-				Pallet::<T>::encoded_ethereum_xcm_call(xcm_domain.clone(), contract_call);
-
-			pallet_xcm_transactor::Pallet::<T>::transact_through_sovereign(
-				T::RuntimeOrigin::root(),
-				// The destination to which the message should be sent
-				xcm_domain.location,
-				// The sender will pay for this transaction.
-				sender,
-				// The currency in which we want to pay fees
-				CurrencyPayment {
-					currency: Currency::AsCurrencyId(xcm_domain.fee_currency),
-					fee_amount: None,
-				},
-				// The call to be executed in the destination chain
-				ethereum_xcm_call,
-				OriginKind::SovereignAccount,
-				TransactWeights {
-					// Convert the max gas_limit into a max transact weight following Moonbeam's
-					// formula.
-					transact_required_weight_at_most: xcm_domain.max_gas_limit * 25_000
-						+ 100_000_000,
-					overall_weight: None,
-				},
-			)?;
-
-			Ok(())
-		}
-	}
+			if let Domain::Centrifuge = destination {
+				return Err("invalid outbound destination".into());
```

Comment:
We _could_ be more specific with this error message and say that we can't send a message to the domain itself